### That's my first experience with pandas, numpy and jupyter notebook, hope to improve someday! Thank you so much for reading.

In [13]:
import pandas as pd
import numpy as np

# Cleaning the data

#### First of all, I will start by changing the date of my csv from a string to a data type. 

In [14]:
d_parser = lambda x: pd.datetime.strptime(x, '%d/%m/%Y')
df = pd.read_csv('data/fase_01.csv', parse_dates=['Data'], date_parser=d_parser)

C:\Users\breno\AppData\Local\Temp\ipykernel_12324\3715823091.py:1: FutureWarning: The pandas.datetime class is deprecated and will be removed from pandas in a future version. Import from datetime module instead.
  d_parser = lambda x: pd.datetime.strptime(x, '%d/%m/%Y')


In [15]:
df.shape

(230, 16)

#### Then exclude dates that pass the current day

In [16]:
filt = df['Data'] <= '2022-02-24'
df = df[filt]

In [17]:
df

Nome do Post       Data  \
0               Amostragem_Imagem_Feed_LinkedIn_150721 2021-07-15   
1         AulõesFuvest_Imagem_Stories_Instagram_201221 2021-12-20   
2      BingoDoEstudante_Vídeo_Stories_Instagram_110821 2021-08-11   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221 2021-12-11   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram... 2021-12-03   
..                                                 ...        ...   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag... 2021-11-11   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta... 2021-11-11   
227      VoltasAulasUSP_Vídeo_Stories_Instagram_160821 2021-08-16   
228         XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721 2021-07-30   
229         XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821 2021-08-05   

                      Criativo Posicionamento Plataforma Formato  Conteúdo  \
0                   Amostragem           Feed   LinkedIn  Imagem       NaN   
1                 AulõesFuvest        Stories  Instagram  Imagem       NaN   
2             BingoDoEstudante        Stories  Instagram   Vídeo       NaN   
3               BoaProvaFuvest        Stories  Instagram  Imagem       NaN   
4        CuriosidadesArcane(1)        Stories  Instagram  Imagem       NaN   
..                         ...            ...        ...     ...       ...   
225  Vale-PresenteResultado(5)        Stories  Instagram   Vídeo       NaN   
226  Vale-PresenteResultado(6)        Stories  Instagram  Imagem       NaN   
227             VoltasAulasUSP         Storie  Instagram   Vídeo       NaN   
228             XadrezNoIME(1)           Feed   Linkedin  Imagem       NaN   
229             XadrezNoIME(2)           Feed   LinkedIn  Imagem       NaN   

     Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
0        NaN         NaN    0.0            108.0         0.0   
1      116.0       119.0    NaN              NaN         NaN   
2       93.0         NaN    0.0              0.0         0.0   
3       75.0        75.0    NaN              NaN         NaN   
4       84.0        88.0    NaN              NaN         NaN   
..       ...         ...    ...              ...         ...   
225     87.0        88.0    NaN              NaN         NaN   
226     89.0        90.0    NaN              NaN         NaN   
227     49.0        49.0    0.0              0.0         0.0   
228      NaN       348.0    0.0             79.0         0.0   
229      NaN       246.0    0.0             33.0         0.0   

     Visualizações 100%  Comentários  Reações  Compartilhamentos  
0                   NaN            0     15.0                0.0  
1                   NaN            0      NaN                1.0  
2                   0.0            0      0.0                0.0  
3                   NaN            0      NaN                0.0  
4                   NaN            1      NaN                1.0  
..                  ...          ...      ...                ...  
225                 NaN            1      NaN                0.0  
226                 NaN            2      NaN                1.0  
227                 0.0            1      0.0                0.0  
228                 NaN            0     18.0                0.0  
229                 NaN            0     10.0                NaN  

[224 rows x 16 columns]

#### As we can see, there are lots of missing values so I will start by counting then and judge if there is relevance on the data, if there is no relevance I will exclude them. I will also count if there are any duplicates.

In [18]:
df.replace('', np.nan, inplace=True)
df.isnull().sum().sort_values(ascending=False)

Conteúdo              224
Visualizações 100%     83
Cliques em Link        55
3-Sec-View             48
Saves                  40
Reações                40
Alcance                23
Impressões              6
Compartilhamentos       5
Nome do Post            0
Data                    0
Criativo                0
Posicionamento          0
Plataforma              0
Formato                 0
Comentários             0
dtype: int64

In [19]:
df.isna().mean().sort_values(ascending=False)

Conteúdo              1.000000
Visualizações 100%    0.370536
Cliques em Link       0.245536
3-Sec-View            0.214286
Saves                 0.178571
Reações               0.178571
Alcance               0.102679
Impressões            0.026786
Compartilhamentos     0.022321
Nome do Post          0.000000
Data                  0.000000
Criativo              0.000000
Posicionamento        0.000000
Plataforma            0.000000
Formato               0.000000
Comentários           0.000000
dtype: float64

In [20]:
df = df.drop(columns='Conteúdo')

In [21]:
df.duplicated(subset=['Nome do Post']).sum()

0

In [22]:
df.shape

(224, 15)

# What is the most reacted post?

In [23]:
df_react = df.loc[:,['Nome do Post', 'Reações']]

In [24]:
df_react = df_react.dropna()
df_react[df_react.Reações == df_react.Reações.max()]

Nome do Post  Reações
109  DiaDaPizza(2)_Vídeo_Stories_Instagram_100721  99999.0

In [25]:
df_react.loc[:,['Reações']].mean()

Reações    568.940217
dtype: float64

#### Here we can cleary see some strange data (so much beyond the mean of the reactions), it don't seems like an outlier but seems like a bug, so what i'm going to do is exclude it and repeat the same process

In [26]:
df_react.drop([109]).max()

Nome do Post    XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821
Reações                                             1899.0
dtype: object

#### Now we can see that the LinkedIn post about chess seems to be the real most reacted post.

# What is the most commented post?

In [27]:
df_coment = df.loc[:, ['Nome do Post', 'Comentários']]

In [28]:
df_coment.loc[:, ['Comentários']].mean()

Comentários    3.410714
dtype: float64

In [29]:
df_coment[df_coment.Comentários == df_coment.Comentários.max()]

Nome do Post  Comentários
217  SorteioValePresente_Imagem_Feed_Instagram_291021          369

#### We can see that the post above is far beyond the mean of the comments, that's because it's a prize draw post that has as requirement for participate that people mark a friend on comments, so I will repeat the process without this post just to see the second most commented post (that don't has this requirement)

In [30]:
df_coment_second = df_coment.drop([217])
df_coment_second[df_coment_second.Comentários == df_coment_second.Comentários.max()]

Nome do Post  Comentários
150  DiaMusicistas_Imagem_Reels_Instagram_221121           79

# What is the most shared post?

In [31]:
df_share = df.loc[:, ['Nome do Post', 'Compartilhamentos']].dropna()
df_share

Nome do Post  Compartilhamentos
0               Amostragem_Imagem_Feed_LinkedIn_150721                0.0
1         AulõesFuvest_Imagem_Stories_Instagram_201221                1.0
2      BingoDoEstudante_Vídeo_Stories_Instagram_110821                0.0
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221                0.0
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...                1.0
..                                                 ...                ...
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...                0.0
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...                0.0
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...                1.0
227      VoltasAulasUSP_Vídeo_Stories_Instagram_160821                0.0
228         XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721                0.0

[219 rows x 2 columns]

In [32]:
df_share.loc[:, ['Compartilhamentos']].mean()

Compartilhamentos    2.182648
dtype: float64

In [33]:
df_share[df_share.Compartilhamentos == df_share.Compartilhamentos.max()]

Nome do Post  Compartilhamentos
214  SemanaPreFuvest_Imagem_Feed_Instagram_060121              200.0

#### The most shared post is about a national exam that occurs in Brazil called ''Fuvest''

# What is the most saved post?

In [34]:
df_save = df.loc[: , ['Nome do Post', 'Saves']]
df_save = df_save.dropna()
df_save.loc[:, ['Saves']].mean()

Saves    1.271739
dtype: float64

In [35]:
df_save[df_save.Saves == df_save.Saves.max()]

Nome do Post  Saves
214  SemanaPreFuvest_Imagem_Feed_Instagram_060121   81.0

#### Again the most saved post is about ''Fuvest''

# What is the total of saves on Instagram?

In [36]:
df_6 = df.loc[:, ['Plataforma', 'Saves']]
df_6 = df_6.dropna()
df_6_grp = df_6.groupby('Plataforma')
df_6_grp.value_counts()

Plataforma  Saves
Instagram   0.0      121
            1.0       12
            2.0        6
            3.0        4
            8.0        3
            7.0        2
            13.0       2
            10.0       1
            12.0       1
            15.0       1
            16.0       1
            81.0       1
LinkedIn    0.0       20
Linkedin    0.0        3
ig          0.0        6
dtype: int64

In [37]:
ig_saves = df_6_grp.get_group('Instagram').loc[:, ['Saves']].sum();
ig_saves

Saves    234.0
dtype: float64

#### We can ignore the group ''ig'' because it has no saves.

# Wich posts reached more users?

In [38]:
df_7 = df.loc[:, ['Nome do Post', 'Alcance']]
df_7 = df_7.dropna()
df_7_sorted = df_7.sort_values(by='Alcance', ascending=False)
df_7_sorted.head(10)

Nome do Post  Alcance
147  DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagra...   6600.0
214       SemanaPreFuvest_Imagem_Feed_Instagram_060121   6058.0
200        Retrospectiva _Vídeo_Reels_Instagram_311221   1173.0
159  FotoConfraternização_Imagem_Feed_Instagram_211221    838.0
150        DiaMusicistas_Imagem_Reels_Instagram_221121    838.0
197      ProcessoSeletivo_Imagem_Feed_Instagram_060921    697.0
183            NovaGestão_Imagem_Feed_Instagram_141221    638.0
171         IMEJrFederada_Imagem_Feed_Instagram_090921    599.0
160      FraseEstatistica_Imagem_Feed_Instagram_281221    556.0
217   SorteioValePresente_Imagem_Feed_Instagram_291021    530.0

#### There are the 10 posts that reached more users.

# How many days had posts?

In [39]:
df_8 = df['Data']
df_8 = df_8.dropna()
df_8_days = df_8.drop_duplicates(keep='last')
df_8_days

0     2021-07-15
1     2021-12-20
3     2021-12-11
9     2021-12-03
13    2022-01-04
         ...    
219   2021-11-08
226   2021-11-11
227   2021-08-16
228   2021-07-30
229   2021-08-05
Name: Data, Length: 88, dtype: datetime64[ns]

#### After removing the duplicates (because there are days with more than 1 post) it's just see the length of the series: 88 days at total.

# How many posts?

In [40]:
df.loc[:, ['Nome do Post']].count()

Nome do Post    224
dtype: int64

#### Just counted the number of datas in the ''name of the post'' series.

# How many users reached on Instagram?

In [41]:
df_9 = df.loc[:, ['Plataforma', 'Alcance']]
df_9 = df_9.dropna()
df_9_grp = df_9.groupby('Plataforma')
df_9_grp.value_counts()

Plataforma  Alcance
Instagram   95.0       6
            87.0       6
            106.0      5
            101.0      5
            97.0       5
                      ..
ig          159.0      1
            127.0      1
            77.0       1
            61.0       1
            310.0      1
Length: 134, dtype: int64

In [42]:
df_9.loc[:, ['Alcance']].sum()

Alcance    44518.0
dtype: float64

#### As we can see LinkedIn doesn't has the attribute ''Reach'' so we can just count the total of Instagram.

# Which is the format of post with most sharings?

In [43]:
df_10 = df.loc[:, ['Formato', 'Compartilhamentos']]
df_10 = df_10.dropna()
df_10_grp = df_10.groupby('Formato')
df_10_grp.value_counts()

Formato  Compartilhamentos
Imagem   0.0                  123
         1.0                   14
         2.0                    7
         4.0                    4
         11.0                   2
         13.0                   1
         18.0                   1
         12.0                   1
         21.0                   1
         100.0                  1
         200.0                  1
         8.0                    1
         3.0                    1
         19.0                   1
Photo    2.0                    1
Video    0.0                    5
Vídeo    0.0                   46
         14.0                   1
imagem   0.0                    4
         2.0                    1
vídeo    0.0                    2
dtype: int64

In [46]:
df_10_grp.get_group('Imagem').sum() + df_10_grp.get_group('imagem').sum() + df_10_grp.get_group('Photo').sum()


Formato              ImagemImagemImagemImagemImagemImagemImagemImag...
Compartilhamentos                                                464.0
dtype: object

In [47]:
df_10_grp.get_group('Video').sum() + df_10_grp.get_group('Vídeo').sum() + df_10_grp.get_group('vídeo').sum()

Formato              VideoVideoVideoVideoVideoVídeoVídeoVídeoVídeoV...
Compartilhamentos                                                 14.0
dtype: object

#### We can see that the images has more sharing than videos, but I will consider the proportion of sharings/(total of posts), because it can be so many more images than videos posted.

In [48]:
df_10_second = df.loc[:, ['Formato']]
df_10_second_grp = df_10_second.groupby('Formato')
df_10_second_grp.value_counts()

Formato
Imagem    162
Photo       1
Video       5
Vídeo      48
imagem      5
vídeo       3
dtype: int64

#### There are 168 photos (imagem + imagem + photo) and 56 videos, so:

In [49]:
Prop_photos = 471/168

In [50]:
Prop_videos = 56/14

In [52]:
Prop_photos

2.8035714285714284

In [53]:
Prop_videos

4.0

#### So we can see that proporcionally videos has more sharing than the photos

# The post about ''Federação da IME Jr'' had better performance on Instagram or LinkedIn?

In [54]:
df_11 = df.loc[:, ['Criativo', 'Plataforma', 'Impressões', 'Saves', 'Cliques em Link', '3-Sec-View', 'Comentários', 'Reações', 'Compartilhamentos' ]]
df_11_grp = df_11.groupby('Criativo')
df_11_fed = df_11_grp.get_group('IMEJrFederada')
df_11_fed

Criativo Plataforma  Impressões  Saves  Cliques em Link  3-Sec-View  \
171  IMEJrFederada  Instagram       669.0    2.0              0.0         0.0   
172  IMEJrFederada   LinkedIn       340.0    0.0             14.0         0.0   

     Comentários  Reações  Compartilhamentos  
171            8     53.0                0.0  
172            2     17.0                1.0

In [55]:
df_11_fed.loc[:, 'Impressões':'Compartilhamentos'].sum(axis=1)

171    732.0
172    374.0
dtype: float64

#### Here I just consider attributes that are mutual for Instagram and Linkedin, but IG keeps with the best performance.

# What is the mean of likes on Instagram? And on Linkedin?

In [56]:
df_12 = df.loc[:, ['Plataforma', 'Reações']].dropna()
df_12_grp = df_12.groupby('Plataforma')
df_12_grp.value_counts()

Plataforma  Reações
Instagram   0.0        111
            37.0         4
            34.0         3
            2.0          2
            43.0         2
            42.0         2
            26.0         2
            32.0         2
            99999.0      1
            460.0        1
            95.0         1
            119.0        1
            88.0         1
            94.0         1
            45.0         1
            10.0         1
            21.0         1
            23.0         1
            25.0         1
            28.0         1
            29.0         1
            35.0         1
            38.0         1
            40.0         1
            48.0         1
            91.0         1
            53.0         1
            69.0         1
            71.0         1
            73.0         1
            74.0         1
            76.0         1
            79.0         1
            1.0          1
            90.0         1
LinkedIn    14.0         4
        

In [58]:
df_12_ig = df_12_grp.get_group('Instagram').sum() + df_12_grp.get_group('ig').sum()
df_12_ig

Plataforma    InstagramInstagramInstagramInstagramInstagramI...
Reações                                                102502.0
dtype: object

In [60]:
df_12_ig_total = df_12_grp.get_group('Instagram').count() + df_12_grp.get_group('ig').count() 
df_12_ig_total

Plataforma    161
Reações       161
dtype: int64

#### Now I'm gonna to use that:
$\bar{x} = \frac{1}{n} \cdot \sum\limits_{i=1}^{n} x_i$

In [62]:
df_ig_mean = (102502.0 - 99999.0)/161-1

In [63]:
df_ig_mean

14.546583850931677

#### Note that I already excluded the bug/strange outlier that we discussed before.

In [64]:
df_12_lk = df_12_grp.get_group('Linkedin').sum() + df_12_grp.get_group('LinkedIn').sum()
df_12_lk

Plataforma    LinkedinLinkedinLinkedinLinkedInLinkedInLinked...
Reações                                                  2183.0
dtype: object

In [65]:
df_12_total = df_12_grp.get_group('Linkedin').count() + df_12_grp.get_group('LinkedIn').count()
df_12_total

Plataforma    23
Reações       23
dtype: int64

In [66]:
df_lk_mean = 2183/23
df_lk_mean

94.91304347826087

# Which month has most reactions?

In [67]:
df_15 = df.loc[:, ['Data', 'Reações']]
df_15 = df_15.dropna()
df_15 = df_15.set_index('Data')
df_15 = df_15['Reações'].resample('M').sum().dropna()

In [ ]:
df_15

In [70]:
df_15 = df_15.reset_index()
df_15.max()

index                       52
Data       2021-12-31 00:00:00
Reações               102605.0
dtype: object

#### We have to consider that that's the month that contais the bug/outlier discussed before, so I will count it as 102605-99999 = 2.606 reaction. Although, it stills the month with most reactions.

# How many and wich posts has more than 15 comments?

In [71]:
df_16 = df.loc[:, ['Nome do Post', 'Comentários']].dropna()
filt = df_16['Comentários'] > 15
df_16[filt]

Nome do Post  Comentários
104          DiaDaFamília_Imagem_Feed_Instagram_150521           20
135        DiaDoProfessor_Imagem_Feed_Instagram_151021           28
150        DiaMusicistas_Imagem_Reels_Instagram_221121           79
159  FotoConfraternização_Imagem_Feed_Instagram_211221           23
183            NovaGestão_Imagem_Feed_Instagram_141221           32
214       SemanaPreFuvest_Imagem_Feed_Instagram_060121           25
217   SorteioValePresente_Imagem_Feed_Instagram_291021          369

In [72]:
df_16[filt]['Nome do Post'].count()

7

#### There are 7 posts with > 15 comments.

# How many posts on feed doesn't has comments? And how many at stories?

In [73]:
df_17 = df.loc[:, ['Posicionamento', 'Comentários']].dropna()
filt = df_17['Comentários'] == 0
df_17 = df_17[filt].groupby('Posicionamento')
df_17.value_counts()

Posicionamento  Comentários
Fed             0                1
Feed            0               23
Storie          0                3
Stories         0              109
Story           0                8
feed            0                1
stories         0               17
dtype: int64

In [74]:
df_17_feed = df_17.get_group('Fed').count() + df_17.get_group('Feed').count() + df_17.get_group('feed').count()
df_17_feed


Posicionamento    25
Comentários       25
dtype: int64

In [75]:
df_17_stories = df_17.get_group('Storie').count() + df_17.get_group('Stories').count() + df_17.get_group('Story').count() + df_17.get_group('stories').count()
df_17_stories

Posicionamento    137
Comentários       137
dtype: int64

#### There are 25 and 137 posts without comments at Feed and Stories, in order.

# How many post on IG feed since 29/10/2021?

In [77]:
df_18 = df.loc[:, ['Data', 'Nome do Post', 'Plataforma', 'Posicionamento']].dropna()
filt = df_18['Data'] >= '2021-10-29'
df_18 = df_18[filt].sort_values(by='Data')
df_18_grp = df_18.groupby('Posicionamento')
df_18_grp.get_group('Feed').count()

Data              6
Nome do Post      6
Plataforma        6
Posicionamento    6
dtype: int64

#### There are 6 since the date.

# How many time each post was seem by users? Which one is the most seem?

#### Here I will consider the 3-Sec-Views and the 100% Views

In [78]:
df_19 = df.loc[:, ['Nome do Post', '3-Sec-View', 'Visualizações 100%']].dropna()
df_19['Total View'] = df_19.agg({'3-Sec-View': 'sum', 'Visualizações 100%': 'sum'})
df_19.max()

Nome do Post          VoltasAulasUSP_Vídeo_Stories_Instagram_160821
3-Sec-View                                                      0.0
Visualizações 100%                                              0.0
Total View                                                      NaN
dtype: object

#### Here we have a problem, there are no 3-Sec-Views or 100% Views, so it's not possible to use those datas, for a good aproximation we can use the reach collected before.

# Which is the post with better reaction performance?

#### To classificate ''Performance'' I usead the Reactions/Reach

In [80]:
df_20 = df.loc[:, ['Nome do Post', 'Alcance', 'Reações']].dropna()
df_20['Performance'] = df_20['Reações'] / df_20['Alcance']
df_20 = df_20.dropna()
df_20[df_20.Performance == df_20.Performance.max()]

Nome do Post  Alcance  Reações  \
109  DiaDaPizza(2)_Vídeo_Stories_Instagram_100721     70.0  99999.0   

     Performance  
109  1428.557143

#### Again excluding the bug/outiler.

In [81]:
df_20 = df_20.drop([109])
df_20[df_20.Performance == df_20.Performance.max()]

Nome do Post  Alcance  Reações  \
177  IsabelleTamarozi(1)_Imagem_Feed_Instagram_280721    279.0    460.0   

     Performance  
177     1.648746

# Which week days there're most reactions?

In [82]:
df_21 = df.loc[: , ['Data', 'Reações']].dropna()
df_21['Data'] = df_21['Data'].dt.day_name()
df_21_grp = df_21.groupby('Data')
df_21_grp.value_counts()

Data       Reações
Friday     0.0        33
           10.0        2
           18.0        1
           32.0        1
           34.0        1
                      ..
Wednesday  88.0        1
           9.0         1
           48.0        1
           32.0        1
           1899.0      1
Length: 71, dtype: int64

In [83]:
df_21_grp.sum()

Reações
Data               
Friday        367.0
Monday        390.0
Saturday   100363.0
Sunday        135.0
Thursday      241.0
Tuesday       449.0
Wednesday    2740.0

#### As we can see Saturday it's the one with most reactions, but we need to remember the bug/outlier, so Saturday will be counted as 100363.0 - 99999.0 = 364.

#### So, the day with most reactions is Wednesday. But it's so far from the other days, just in case I will see which one is that post.

In [84]:
df_21_second = df.loc[: , ['Nome do Post','Data', 'Reações']].dropna()
df_21_second['Data'] = df_21_second['Data'].dt.day_name()
df_21_grp_second = df_21_second.groupby('Data')
df_21_grp_second.max()

Nome do Post  Reações
Data                                                                 
Friday            XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721     94.0
Monday         VoltasAulasUSP_Vídeo_Stories_Instagram_160821     79.0
Saturday   PrintReuniãoGeral_Imagem_Stories_Instagram_290521  99999.0
Sunday       SemanaMeioAmbiente_Imagem_Feed_Instagram_060621     76.0
Thursday          XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821     73.0
Tuesday          SetembroAmarelo_Imagem_Feed_LinkedIn_210921    119.0
Wednesday       SemanaPreFuvest_Imagem_Feed_Instagram_060121   1899.0

#### As we can see it's a post about Fuvest, so it isn't a bug, just a post that hitted and it's responsable for more than 50% of the reactions at the day.

# What is the total of impressions that the posts with ''DiaDoDesafio'' has?

In [85]:
df_22 = df.loc[:, ['Criativo', 'Impressões']].dropna()
df_22 = df_22[df_22['Criativo'].str.contains('DiaDoDesafio')]
df_22

Criativo  Impressões
120   DiaDoDesafio(1)       194.0
121  DiaDoDesafio(10)       116.0
122   DiaDoDesafio(2)       172.0
123   DiaDoDesafio(3)       156.0
124   DiaDoDesafio(4)       150.0
125   DiaDoDesafio(5)       139.0
126   DiaDoDesafio(6)       136.0
127   DiaDoDesafio(7)       132.0
128   DiaDoDesafio(8)       128.0
129   DiaDoDesafio(9)       120.0

In [86]:
df_22['Impressões'].sum()

1443.0

# How many like the post about ''Isabelle Tamarozi'' has? Was it a good post? Why?

In [87]:
df_20[df_20.Performance == df_20.Performance.max()]

Nome do Post  Alcance  Reações  \
177  IsabelleTamarozi(1)_Imagem_Feed_Instagram_280721    279.0    460.0   

     Performance  
177     1.648746

#### As we can see it has 460 reactions (considering that reactions = likes). So it was a really great post with a really nice performance. 

# What is the name of the best story? Why?

In [93]:
df_24 = df.loc[:, ['Nome do Post', 'Posicionamento', 'Alcance', 'Reações', 'Impressões']].dropna()
df_24 = df_24[df_24['Posicionamento'].str.contains('St', 'st')]
df_24['Performance'] = df_24['Reações'] / df_24['Alcance']
df_24

Nome do Post Posicionamento  Alcance  \
14   DadosPopulação(1)_Vídeo_Stories_Instagram_170721        Stories    120.0   
15   DadosPopulação(2)_Vídeo_Stories_Instagram_170721        Stories    106.0   
16   DadosPopulação(3)_Vídeo_Stories_Instagram_170721        Stories     99.0   
17   DadosPopulação(4)_Vídeo_Stories_Instagram_170721        Stories     97.0   
18   DadosPopulação(5)_Vídeo_Stories_Instagram_170721        Stories     93.0   
..                                                ...            ...      ...   
203          Round6(3)_Vídeo_Stories_Instagram_191021        Stories     97.0   
204          Round6(4)_Vídeo_Stories_Instagram_191021        Stories     97.0   
205          Round6(5)_Vídeo_Stories_Instagram_191021        Stories     96.0   
206          Round6(6)_Vídeo_Stories_Instagram_191021        Stories     95.0   
227     VoltasAulasUSP_Vídeo_Stories_Instagram_160821         Storie     49.0   

     Reações  Impressões  Performance  
14       0.0       123.0     0.000000  
15       0.0       109.0     0.000000  
16       0.0       100.0     0.000000  
17       0.0        98.0     0.000000  
18       0.0        95.0     0.000000  
..       ...         ...          ...  
203      0.0        98.0     0.000000  
204      0.0        97.0     0.000000  
205      2.0        96.0     0.020833  
206      1.0        95.0     0.010526  
227      0.0        49.0     0.000000  

[104 rows x 6 columns]

In [94]:
df_24 = df_24.replace(np.inf, np.nan)
df_24[df_24.Performance == df_24.Performance.max()]

Nome do Post Posicionamento  Alcance  \
109  DiaDaPizza(2)_Vídeo_Stories_Instagram_100721         Storie     70.0   

     Reações  Impressões  Performance  
109  99999.0        70.0  1428.557143

#### Again our problem with the bug/outlier, I will exclude it and continue.

In [95]:
df_24 = df_24.drop([109])
df_24[df_24.Performance == df_24.Performance.max()]

Nome do Post Posicionamento  Alcance  \
118  DiaDoAtletismo_Imagem_Stories_Instagram_091021         Storie     87.0   

     Reações  Impressões  Performance  
118      2.0        88.0     0.022989

#### Another way to see if is a story nice it's looking only to it reach:

In [96]:
df_24[df_24.Alcance == df_24.Alcance.max()]

Nome do Post Posicionamento  \
147  DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagra...          Story   

     Alcance  Reações  Impressões  Performance  
147   6600.0      0.0        67.0          0.0

#### But there's a problem, we can see that the reach > impressions, that's impossible on IG, so I will exclude that data and continue.

In [97]:
df_24 = df_24.drop([147])
df_24[df_24.Alcance == df_24.Alcance.max()]

Nome do Post Posicionamento  Alcance  \
120  DiaDoDesafio(1)_Imagem_Stories_Instagram_260521        Stories    189.0   

     Reações  Impressões  Performance  
120      0.0       194.0          0.0

#### PS: That problem can affect previous datas collected